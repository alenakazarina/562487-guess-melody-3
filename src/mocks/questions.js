import {QuestionType} from '../const';

const questions = [
  {
    type: QuestionType.GENRE,
    genre: `рэгги`,
    answers: [{
      src: `https://res.cloudinary.com/df6cuiopc/video/upload/v1581414459/music/track-08.mp3`,
      genre: `pop`,
    }, {
      src: `https://res.cloudinary.com/df6cuiopc/video/upload/v1581414459/music/track-07.mp3`,
      genre: `rnb`,
    }, {
      src: `https://res.cloudinary.com/df6cuiopc/video/upload/v1581414459/music/track-05.mp3`,
      genre: `dubstep`,
    }, {
      src: `https://res.cloudinary.com/df6cuiopc/video/upload/v1581414459/music/track-06.mp3`,
      genre: `reggae`
    }]
  },
  {
    type: QuestionType.ARTIST,
    song: {
      artist: `Rihanna`,
      src: `https://res.cloudinary.com/df6cuiopc/video/upload/v1581414089/music/track-04.mp3`,
    },
    answers: [{
      picture: `https://res.cloudinary.com/df6cuiopc/image/upload/v1581416516/avatars/artist-01.jpg`,
      artist: `Rita Ora`,
    }, {
      picture: `https://res.cloudinary.com/df6cuiopc/image/upload/v1581416516/avatars/artist-02.jpg`,
      artist: `Rihanna`,
    }, {
      picture: `https://res.cloudinary.com/df6cuiopc/image/upload/v1581416516/avatars/artist-03.jpg`,
      artist: `Beyonce`,
    }]
  }
];

export {questions, QuestionType};
