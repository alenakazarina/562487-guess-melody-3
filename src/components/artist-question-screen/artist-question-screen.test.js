import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from './artist-question-screen';
import {questions} from '../../mocks/questions';

describe(`ArtistQuestionScreen`, () => {
  it(`should render ArtistQuestionScreen correctly`, () => {
    const tree = renderer.create(
        <ArtistQuestionScreen
          question={questions[1]}
          onAnswer={()=>{}}
          renderPlayer={()=>{}}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
