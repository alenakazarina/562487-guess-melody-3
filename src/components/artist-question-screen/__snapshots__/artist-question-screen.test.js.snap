// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen should render ArtistQuestionScreen correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Rita Ora"
          className="artist__picture"
          src="https://res.cloudinary.com/df6cuiopc/image/upload/v1581416516/avatars/artist-01.jpg"
        />
        Rita Ora
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Rihanna"
          className="artist__picture"
          src="https://res.cloudinary.com/df6cuiopc/image/upload/v1581416516/avatars/artist-02.jpg"
        />
        Rihanna
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Beyonce"
          className="artist__picture"
          src="https://res.cloudinary.com/df6cuiopc/image/upload/v1581416516/avatars/artist-03.jpg"
        />
        Beyonce
      </label>
    </div>
  </form>
</section>
`;
